Storage/src/Connection/RetryTrait.php

```diff
@@ -161,32 +198,38 @@ private function isPreConditionSupplied($methodName, $preConditions)
      * @param \Exception $exception The exception object received
      * while sending the request.
      * @param int $currentAttempt Current retry attempt.
-     * @param boolean $isIdempotent
-     * @param boolean $preconditionNeeded
-     * @param boolean $preconditionSupplied
+     * @param bool $isIdempotent
+     * @param bool $preconditionNeeded
+     * @param bool $preconditionSupplied
      * @param int $maxRetries
-     * @return boolean
+     * @return bool
      */
     private function retryDeciderFunction(
         \Exception $exception,
         $currentAttempt,
         $isIdempotent,
         $preconditionNeeded,
         $preconditionSupplied,
-        $maxRetries
+        $maxRetries,
+        $retryStrategy
     ) {
         // No retry if maxRetries reached
         if ($maxRetries <= $currentAttempt) {
             return false;
         }
 
         $statusCode = $exception->getCode();
-
         // Retry if the exception status code matches
         // with one of the retriable status code and
         // the operation is either idempotent or conditionally
         // idempotent with preconditions supplied.
-        if (in_array($statusCode, $this->httpRetryCodes)) {
+        if (in_array($statusCode, self::$httpRetryCodes)) {
+            switch ($retryStrategy) {
+                case self::$RETRY_STRATEGY_ALWAYS:
+                    return true;
+                case self::$RETRY_STRATEGY_NEVER:
```

Comment:
So, the `never` case still need to go over the status code check. Perhaps best to break the switch statement now.

Reply:
You mean we should do it like this?🤔

```php
private function retryDeciderFunction(
        \Exception $exception,
        $currentAttempt,
        $isIdempotent,
        $preconditionNeeded,
        $preconditionSupplied,
        $maxRetries,
        $retryStrategy
    ) {
        // No retry if maxRetries reached
        if ($maxRetries <= $currentAttempt) {
            return false;
        }
        if($retryStrategy == self::$RETRY_STRATEGY_NEVER) {
            return false;
        }

        $statusCode = $exception->getCode();
        // Retry if the exception status code matches
        // with one of the retriable status code and
        // the operation is either idempotent or conditionally
        // idempotent with preconditions supplied.

        if (in_array($statusCode, self::$httpRetryCodes)) {
            if($retryStrategy == self::$RETRY_STRATEGY_ALWAYS) {
                return true;
            } else if ($isIdempotent) {
                return true;
            } elseif ($preconditionNeeded) {
                return $preconditionSupplied;
            }
        }

        return false;
    }
```

Reply:
Yup, this seems correct.

Reply:
Okay, implemented in commit [d69282c](https://github.com/googleapis/google-cloud-php/pull/5731/commits/d69282c93d79f6a469bd17d4454bce5343fafdd6)